### PR TITLE
add copt solver

### LINKED ISSUE #4: add copt solver
see https://github.com/openopt/copt/blob/master/examples/plot_accelerated.py

### PR DESCRIPTION
**Pull request by agramfort**
closes #4

### CODE REVIEW

File: solvers/copt.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        n_features = X.shape[1]
+
+        x0 = np.zeros(n_features)
+        if n_iter == 0:
+            self.beta = x0
+            return
+
+        f = copt.loss.LogLoss(X, y)
+        g = copt.penalty.L1Norm(self.lmbd / X.shape[0])
```
Comment (tomMoral) [inline solvers/copt.py]: It feels weird to have both `cp` and `copt`.
```suggestion
        f = cp.loss.LogLoss(X, y)
        g = cp.penalty.L1Norm(self.lmbd / X.shape[0])
```

File: solvers/copt.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                x0,
+                g.prox,
+                step=step,
+                tol=0,
+                max_iter=n_iter,
+                jac=True,
+                accelerated=self.accelerated,
+            )
+        elif solver == 'saga':
+            step_size = 1.0 / (3 * f.max_lipschitz)
```
Comment (tomMoral) [inline solvers/copt.py:84]: What is the difference with `lipschitz`?

File: solvers/copt.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            result = cp.minimize_saga(
+                f.partial_deriv,
+                X,
+                y,
+                np.zeros(n_features),
```
Comment (tomMoral) [inline solvers/copt.py]:
```suggestion
               x0,
```

File: solvers/copt.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                max_iter=n_iter,
+            )
+        else:
+            assert solver == 'svrg'
+            step_size = 1.0 / (3 * f.max_lipschitz)
+            result = cp.minimize_svrg(
+                f.partial_deriv,
+                X,
+                y,
+                np.zeros(n_features),
```
Comment (tomMoral) [inline solvers/copt.py]:
```suggestion
                x0,
```

**Review by tomMoral — APPROVED** [review]
I have to say the way we use `copt` API looks confusing.
Would it be possible to have a more uniform way of calling each algorithm?

File: solvers/copt.py (hunk 2 above)
Comment (agramfort) [reply solvers/copt.py:84]: max_lipschitz is the max lipschitz constant over each terms in the finite sum in the ERM

**Comment by agramfort** [comment]
greeen !!!!

**Pull request merged**